tests/gtest/test_verilog.cpp

```diff
@@ -345,6 +345,26 @@ TEST(VerilogTests, TestInlineIndex) {
   deleteContext(c);
 }
 
+TEST(VerilogTests, TestCompileGuard) {
```

Comment:
can we add a simpler test? this has a bunch of inline verilog stuff. i think a simple test with just an instance wrapped in compile guard would be useful.